web/modules/custom/dpl_related_content/dpl_related_content.module

```diff
@@ -0,0 +1,292 @@
+  $subquery->fields('es', ['id']);
+
+  // Join the term field tables.
+  // @todo this only looks at the values on the eventseries.
+  // Ideally, we'd want to also look if values exist on the instance level.
+  $subquery->leftJoin('eventseries__field_tags', 'es_tags', 'es.id = es_tags.entity_id');
+  $subquery->leftJoin('eventseries__field_categories', 'es_cats', 'es.id = es_cats.entity_id');
+  $subquery->leftJoin('eventseries__field_branch', 'es_bra', 'es.id = es_bra.entity_id');
+
+  if (!empty($tags) || !empty($categories) || !empty($branches)) {
+    $or_group = $subquery->orConditionGroup();
+
+    // To avoid errors, related to the OR GROUP only containing one condition,
+    // we'll add a fake condition to fill out a possible empty space.
+    $or_group->condition('title', 'ALWAYS_FALSE');
+
+    if (!empty($tags)) {
+      $or_group->condition('es_tags.field_tags_target_id', $tags, 'IN');
+    }
+
+    if (!empty($categories)) {
+      $or_group->condition('es_cats.field_categories_target_id', $categories, 'IN');
+    }
+
+    if (!empty($branches)) {
+      $or_group->condition('es_bra.field_branch_target_id', $branches, 'IN');
+    }
+
+    $subquery->condition($or_group);
+  }
+
+  $subquery->distinct(TRUE);
+
+  // Main query to select eventinstance ids, joining with
+  // eventinstance_field_data for condition fields.
+  $query = $connection->select('eventinstance_field_data', 'eid');
+  $query->join('eventinstance', 'ei', 'ei.id = eid.id');
+  $query->addField('eid', 'id', 'eventinstance_id');
+
+  if (!empty($tags) || !empty($categories) || !empty($branches)) {
+    // Use the subquery to filter by eventseries_id.
+    $query->condition('eid.eventseries_id', $subquery, 'IN');
+  }
+
+  $query->condition('ei.uuid', $current_uuid, '<>');
+
+  // We only want events in the future (e.g. - active events).
+  $query->condition('eid.date__value', $formatted_date, '>=');
+  $query->orderBy('eid.date__value', 'ASC');
+  $query->range(0, 16);
+
+  // Add a GROUP BY clause to make results distinct by eventseries_id.
+  $query->groupBy('eid.eventseries_id');
+
+  // Execute the query and return ids.
+  $result = $query->execute();
+
+  return $result?->fetchCol() ?? [];
+}
+
+/**
+ * Get matching node IDs.
+ *
+ * Allows for passing along various term IDs, that we look for in an OR group.
+ *
+ * @param string|null $current_uuid
+ *   A possible entity UUID, that will not get included in results.
+ * @param array<int> $tags
+ *   Tag term IDs, to look for.
+ * @param array<int> $categories
+ *   Category term IDs, to look for.
+ *
+ * @return array<int|string>
+ *   Matching article node IDs
+ */
+function _dpl_related_content_get_article_ids(?string $current_uuid, array $tags = [], array $categories = []): array {
+  $query = \Drupal::entityQuery('node')->accessCheck(TRUE)
+    ->condition('type', 'article')
+    ->condition('uuid', $current_uuid, '<>')
+    ->sort('created', 'DESC')
```

Comment:
This does sorting but there is no mention about if/why this is important/necessary.

Reply:
As far as I can tell has been changed to a property but I do not see an explaination.

With the sorting being `created`, `DESC` then we end up with the latest content first. I guess that is the priotization.